content/post/gld.md

```diff
@@ -0,0 +1,276 @@
+---
+author: vadim
+date: 2018-04-17
+title: "Detecting licenses in code with Go and ML"
+image: /post/gld/wthpl.png
+description: "Detecting the license of an open source projects is harder than it seems. We have created go-license-detector, a Go library and command line application to solve that task."
+categories: ["science", "technical"]
+---
+
+While working on [Public Git Archive](https://github.com/src-d/datasets/PublicGitArchive), we
+thought that it would be handy to include the license of each project in the index file, so that
+people could easily filter "grey" repositories without a clear license. Besides, we were
+curious about the licenses distribution. GitHub already detects licenses by leveraging
+the Ruby library [benbalter/licensee](https://github.com/benbalter/licensee) and the easy solution was
+to query the GitHub API. However, we were not satisfied with
+its detection quality: many projects which actually contain the license file in a non-standard
+format are missed, and some are misclassified. This is how
+[go-license-detector](https://github.com/src-d/go-license-detector) was born.
+
+The goals were defined from the very beginning:
+
+1. Favor false positives over false negatives (target data mining instead of compliance).
+2. Perform fast.
+3. Detect as many licenses as possible on the [hand-collected dataset of 1,000 top-starred repositories
```

Comment:
I would make it more clear that this 1k repos dataset was manually checked and labelled.